### PR TITLE
Delete all dockershim presubmits

### LINKED ISSUE #24620: Move docker-based node presubmits to CRI
**What should be cleaned up or changed**:

SIG Node has a number of presubmit jobs that need to be migrated off Docker.

I suggest we declare bankruptcy, remove all Docker-based jobs, and add jobs back under the CRI as needed.

/sig node

### PR DESCRIPTION
**Pull request by matthyx**
Relates to #24620 
/cc @ehashman

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.20.yaml [hunk 1]
```diff
@@ -1067,43 +1067,6 @@ presubmits:
...
             memory: 6Gi
-  - always_run: true
-    branches:
-    - release-1.20
-    cluster: k8s-infra-prow-build
-    labels:
-      preset-k8s-ssh: "true"
-      preset-service-account: "true"
-    max_concurrency: 12
-    name: pull-kubernetes-node-e2e
```
Comment (matthyx) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.20.yaml:1078]: Not sure it's good to remove it from 1.20 and 1.21 but if not config tests fail.

**Review by ehashman — CHANGES_REQUESTED** [review]
Please don't delete any of the presubmits on previous release branches, this should only apply to master.

**Comment by matthyx** [comment]
> Please don't delete any of the presubmits on previous release branches, this should only apply to master.

Yeah this is what I thought... I need to find a way to make tests pass if I keep them in the 1.2x branches.
I think deleting them from 1.23 is OK since we removed dockershim at that point?

**Comment by ehashman** [comment]
No, 1.23 shipped with dockershim. Only 1.24+ is affected

**Comment by matthyx** [comment]
> No, 1.23 shipped with dockershim. Only 1.24+ is affected

Ok, I will correct tomorrow and see how to make the tests pass. Thanks for your help Elana :-)

File: config/testgrids/config.yaml [hunk 2]
```diff
@@ -77,6 +77,18 @@ test_groups:
 - name: test_pull_request_crio_e2e_rhel
   gcs_prefix: origin-federated-results/pr-logs/directory/test_pull_request_crio_e2e_rhel
 
+# dockershim
+- name: pull-kubernetes-node-e2e
+  gcs_prefix: kubernetes-jenkins/pr-logs/directory/pull-kubernetes-node-e2e
+- name: pull-kubernetes-node-e2e-alpha
+  gcs_prefix: skip
```
Comment (matthyx) [inline config/testgrids/config.yaml:84]: need to set to non-existing prefix for config/tests/testgrids/config_test.go:TestKubernetesProwInstanceJobsMustHaveMatchingTestgridEntries

**Comment by dims** [comment]
/approve

will leave `/lgtm` to either @ehashman or @SergeyKanzhelev

File: config/testgrids/config.yaml (hunk 2 above)
Comment (ehashman) [reply config/testgrids/config.yaml:84]: This is weird, we're removing them so why would we have to add these to testgrid configs?

Comment (matthyx) [reply config/testgrids/config.yaml:84]: If you want I could take some time to document it in a comment...

Comment (BenTheElder) [reply config/testgrids/config.yaml:84]: pretty sure the testgrid annotations on the prowjobs could be moved to the most recent branch that still has these jobs (most recent so this doesn't come back as old branches are removed from config)

this part:
```yaml
    annotations:
      testgrid-dashboards: sig-node-presubmits
      testgrid-tab-name: pr-node-kubelet-serial-kubetest2
```

Comment (matthyx) [reply config/testgrids/config.yaml:84]: Sorry @BenTheElder I don't understand... test-infra needs to be always on the master branch, but the jobs corresponding to k/k 1.21 are defined in ‎config/jobs/kubernetes/sig-release/release-branch-jobs/1.21.yaml‎ for example, aren't they?
So I need to define the testgrid entries somewhere on test-infra master, no?

**Review by ehashman — COMMENTED** [review]
/lgtm

**Pull request merged**